flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/metrics/NumberOfFullRestartsGauge.java

```diff
@@ -42,6 +42,6 @@ public NumberOfFullRestartsGauge(ExecutionGraph executionGraph) {
 
 	@Override
 	public Long getValue() {
-		return eg.getNumberOfFullRestarts();
```

Comment:
It would be good to deprecate or even remove this method. I think `testRestartWithSlotSharingAndNotEnoughResources()` is problematic because it relies on multi-threading.

Reply:
done.